internal/coreinternal/attraction/attraction.go

```diff
@@ -172,8 +172,8 @@ type AttrProc struct {
 // and returns a AttrProc to be used to process attributes.
 // An error is returned if there are any invalid inputs.
 func NewAttrProc(settings *Settings) (*AttrProc, error) {
-	var attributeActions []attributeAction
-	for i, a := range settings.Actions {
+	attributeActions := make([]attributeAction, 0, len(settings.Actions))
+	for _, a := range settings.Actions {
```

Comment:
Looks like `i` is still needed
```suggestion
	for i, a := range settings.Actions {
```